surefire-api/src/main/java/org/apache/maven/surefire/booter/CommandReader.java

```diff
@@ -412,6 +412,10 @@ public void run()
                                 CommandReader.this.wakeupIterator();
                                 callListeners( command );
                                 break;
+                            case BYE_ACK:
```

Comment:
It is not very elegant solution.
In the `ForkedBooter` the listener is already registered `commandReader.addByeAckListener( new CommandListener()` , see the line 356. See the next comment.

Reply:
One thing I do not understand is why this change cause IT failure on macOS.

Reply:
It would be great to reproduce it with several runs of the build.